src/test/ui/span/missing-unit-argument.stderr

```diff
@@ -2,88 +2,101 @@ error[E0061]: this enum variant takes 1 argument but 0 arguments were supplied
   --> $DIR/missing-unit-argument.rs:11:33
    |
 LL |     let _: Result<(), String> = Ok();
-   |                                 ^^-- supplied 0 arguments
+   |                                 ^^--
+   |                                 |
+   |                                 an argument of type `()` is missing
    |
-help: expected the unit value `()`; create it with empty parentheses
+help: provide the argument
    |
 LL |     let _: Result<(), String> = Ok(());
-   |                                    ++
+   |                                 ~~~~~~
 
 error[E0061]: this function takes 2 arguments but 0 arguments were supplied
   --> $DIR/missing-unit-argument.rs:12:5
    |
 LL |     foo();
-   |     ^^^-- supplied 0 arguments
+   |     ^^^--
    |     |
-   |     expected 2 arguments
+   |     an argument of type `()` is missing
+   |     an argument of type `()` is missing
    |
```

Comment:
Should we special case this to provide a single `span_label` saying "two arguments of type `()` and `()` are missing"?

Reply:
Good idea. Might be worth thinking about consecutive missing args in general:
```
an argument of type `X` is missing
``` 
```
two arguments of type `X` and `Y` are missing
```
```
three arguments of type `X`, `Y`, and `Z` are missing
```
```
multiple missing arguments
```
(the last one is still useful because of the "function defined here" note

I think we only need to fall back here for *consecutive* arguments; if missing args are interspersed by others, the spans will be different and it won't be confusing

We also can probably have a better span here: ideally probably an "empty" span between the parantheses.

Reply:
What I like about the current output is that it is simple and effective regardless of count: "expected 2 arguments"/"supplied 0 arguments" doesn't need that much logic to work well in all cases.

An empty span between the parentheses will show up in a problematic way, because it would display as an underline under the closing `)`, and we wouldn't want that. But it can point at the `()`, like it does now.

Reply:
I'll play around a bit with this. I though you could point at an "empty" span, with just like a line.